### ISSUE TITLE
Question: did you benchmark it?

### ISSUE THREAD

**Issue by AlexGalays** [dai-shi/react-tracked#9]
Hello! the idea behind react-tracked seems interesting.

You claim the performances are better but did you actually verify that? Aren't you creating a completely new proxy on each update?

Cheers

**Comment by dai-shi** [comment]
Hi,

The latest benchmark result at the moment is here: https://github.com/dai-shi/react-tracked/issues/1#issuecomment-519509857
Compare `react-tracked-useSelector` and `react-tracked-useTrackedState` too see the tracking overhead.

> creating a completely new proxy on each update

Yeah, that's the point, because we don't know when render process ends.
It's optimized with per-hook cache, so it won't create a new proxy for the same object.
This means immutable state update is important.
Related discussion: https://github.com/dai-shi/react-tracked/issues/7#issuecomment-518426817

**Comment by AlexGalays** [comment]
Ok thanks, guess we can close this then :)

It seems useSelector is not that much faster and is quite a bit less ergonomic, what do you think?

**Comment by dai-shi** [comment]
Theoretically, useSelector is faster than useTrackedState if a selector is properly implemented.

The code in js-framework-benchmark is very simple and we don't see a big difference. You can see useSelector is faster in "select row."

Here's another benchmark for reactive-react-redux, as a reference.
https://github.com/dai-shi/reactive-react-redux/issues/32#issuecomment-515931658
Those benchmarks are with extreme usage, and especially for "stockticker" useTrackedState is slower.

In summary, useTrackedState is as fast as useSelector for most cases. In some cases, useTrackedState is slower. I don't have a benchmark, but if a developer wrote a selector improperly (which is likely), useSelector would be much slower.

**Comment by dai-shi** [comment]
Forgot to respond this part:

> less ergonomic

Totally. It should be much more intuitive for most of the cases. 
Some edge cases exist, though, as written in the limitations section. It may occur if you deal with object ref equality.
